utils/download_audio.sh

```diff
@@ -1,36 +1,56 @@
 #!/usr/bin/env bash
 # Copyright 2021  Jiayu Du
 #                 Seasalt AI, Inc (Author: Guoguo Chen)
-
+#                 Tsinghua University (Author: Shuzhou Chai) 
 
 set -e
 set -o pipefail
 
-if [ $# -ne 1 ]; then
-  echo "Usage: $0 <gigaspeech-dataset-dir>"
-  echo " e.g.: $0 /disk1/audio_data/gigaspeech"
+
+subset=XL
+
+. ./utils/parse_options.sh || exit 1;
+
+if [ $# -ne 2 ]; then
+  echo "Usage: $0 [options] <gigaspeech-dataset-dir> <GigaSpeech-json> "
+  echo " e.g.: $0 --subset XL /disk1/audio_data/gigaspeech /disk1/GigaSpeech.json"
   echo ""
   echo "This script downloads the entire GigaSpeech audio collection. We"
   echo "suggest having at least 600G of free space in local dir."
   echo "If audios are updated, you can just re-run this script for "
   echo "incremental downloading, downloader will only download updates"
+  echo "You may choose a specific subset to download, including {XS}, {S},"
+  echo "{M}, {L}, {XL}, {DEV}, {TEST} and the default subset is XL."
   echo "After downloading, you may run utils/check_audio_consistency.sh"
   echo "to make sure everything is consistent with official release."
   exit 1
 fi
 
-. ./env_vars.sh || exit 1
+. ./env_vars.sh || exit 1;
+
+
 if [ -z "${GIGA_SPEECH_RELEASE_URL}" ]; then
   echo "ERROR: env variable GIGA_SPEECH_RELEASE_URL is empty(check env_vars.sh?)"
   exit 1
 fi
 
 gigaspeech_dataset_dir=$1
+GigaSpeech_json=$2
 
 [ `uname -s` == 'Linux' ] && ossbin=tools/downloader/ossutil64
 [ `uname -s` == 'Darwin' ] && ossbin=tools/downloader/ossutilmac64
 
-$ossbin -c SAFEBOX/aliyun_ossutil.cfg \
-  cp -ur ${GIGA_SPEECH_RELEASE_URL}/audio/ $gigaspeech_dataset_dir/audio || exit 1
+if [ "$subset" == "XL" ];then 
+  $ossbin -c SAFEBOX/aliyun_ossutil.cfg \
+    cp -ur ${GIGA_SPEECH_RELEASE_URL}/audio/ $gigaspeech_dataset_dir/audio/ || exit 1
+else
+  python3 utils/extract_subset.py \
+    $subset $GigaSpeech_json || exit ;
+  for line in `cat log/subset_path`
```

Comment:
This assumes that utils/extract_subset.py will create a folder called `log` and then a file `subset_path`. It is not a good practice in two ways:

1. `log/subset_path` is hardcoded in utils/extract_subset.py which utils/download_audio.sh can't control
2. `subset_path` does not bear any subset information from its name

I'd suggest you explicitly write out the subset file, e.g., something like

```bash
  python3 utils/extract_subset.py $subset $GigaSpeech_json \
    $gigaspeech_dataset_dir/subset_path_${subset} || exit ;
```

And in `utils/extract_subset.py`, you write directly to `$gigaspeech_dataset_dir/subset_path_${subset}`